cert-generation/main.go

```diff
@@ -50,66 +87,12 @@ func getSecureServiceSpec(namespace string) *corev1.Service {
 			Selector: label,
 		},
 	}
-
-	return service
-}
-
-func getSecureConfigMapSpec(namespace string) *corev1.ConfigMap {
-	return &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      certConfigMapName,
-			Namespace: namespace,
-			Annotations: map[string]string{
-				"service.beta.openshift.io/inject-cabundle": "true",
-			},
-		},
+	_, err = clientset.CoreV1().Services(namespace).Create(service)
+	if err != nil {
+		os.Exit(1)
 	}
-}
-
-func getCertVolume() corev1.Volume {
-	return corev1.Volume{
-		Name: certVolumeName,
-		VolumeSource: corev1.VolumeSource{
-			Projected: &corev1.ProjectedVolumeSource{
-				Sources: []corev1.VolumeProjection{
-					{
-						ConfigMap: &corev1.ConfigMapProjection{
-							LocalObjectReference: corev1.LocalObjectReference{
-								Name: certConfigMapName,
-							},
-							Items: []corev1.KeyToPath{
-								{
-									Key:  "service-ca.crt",
-									Path: "./ca.crt",
-								},
-							},
-						},
-					},
-					{
-						Secret: &corev1.SecretProjection{
-							LocalObjectReference: corev1.LocalObjectReference{
-								Name: certSecretName,
-							},
-						},
-					},
-				},
-			},
-		},
-	}
-}
-
-func getCertVolumeMount() corev1.VolumeMount {
-	return corev1.VolumeMount{
-		Name:      certVolumeName,
-		MountPath: webhookServerCertDir,
-		ReadOnly:  true,
-	}
-}
 
-func getPort() corev1.ContainerPort {
-	port := int32(8443)
-	return corev1.ContainerPort{
-		ContainerPort: port,
-		Name:          webhookServerName,
+	// Wait until its cleaned up by the workspace-controller
+	for {
```

Comment:
Again adding logs could be useful like `I've done my work and waiting to be utilized. Don't feel sorry for me I was initially designed to die if everything is fine` :smile: